src/gluonts/model/wavenet/_estimator.py

```diff
@@ -275,9 +276,18 @@ def train(
             bin_edges, pred_length=self.train_window_length
         )
 
+        # ensure that the training network is created within the same MXNet
+        # context as the one that will be used during training
+        with self.trainer.ctx:
+            params = self._get_wavenet_args(bin_centers)
+            params.update(pred_length=self.train_window_length)
+            trained_net = WaveNet(**params)
+
+        input_names = get_hybrid_forward_input_names(trained_net)
+
         training_data_loader = TrainDataLoader(
             dataset=training_data,
-            transform=transformation,
+            transform=transformation + SelectFields(input_names),
```

Comment:
This now hides what a batch looks like. I understand that that is the point, but maybe that should be mentioned somehow?

Reply:
I'm not sure how this can be clarified though: this was already a bit implicit, and now we just have one more step at this level (instead of in the trainer). Maybe to understand: in what way were things more explicit before?